packages/trace-viewer/src/ui/workbenchLoader.tsx

```diff
@@ -72,6 +72,29 @@ export const WorkbenchLoader: React.FunctionComponent<{
     document.addEventListener('paste', listener);
     return () => document.removeEventListener('paste', listener);
   });
+  React.useEffect(() => {
+    const listener = async (e: MessageEvent) => {
+      if (!e.data.b64trace)
+        return;
+
+      const bytes = atob(e.data.b64trace);
+      const arrayBuffer = new ArrayBuffer(bytes.length);
+      const uint8Array = new Uint8Array(arrayBuffer);
+
+      for (let i = 0; i < bytes.length; i++)
+        uint8Array[i] = bytes.charCodeAt(i);
+
+      const traceBlob = new Blob([uint8Array], { type: 'application/zip' });
```

Comment:
JavaScripts `window.postMessage` API allows sending Blobs from one frame to another. Why does it need to be serialised as base64? My intuition would say accepting a Blob is more reasonable or do I miss something?

Reply:
I could be. `base64` is a naive requirement in the Allure context because we have two ways of the trace file representation – real file (e.g. blob) and base64 encoded file (to be able to access the file in single-file report mode). Of course, we can do a conversion to Blob, no problem, like you said – it's more reasonable because it doesn't lock the solution to our specific use-case

Reply:
Changed to the Blob 👍